src/components/carsoulSection/carsoureItem.js

```diff
@@ -0,0 +1,162 @@
+      id: 5,
+    },
+    {
+      cover:
+        'https://s-media-cache-ak0.pinimg.com/736x/78/c5/f0/78c5f0de85565a0fd4a911f2681f56e3--book-cover-art-cover-books.jpg',
+      authorName: 'Alice Mayer',
+      bookName: 'The Shadows: A Novel',
+      type: 'formate:Hardback',
+      price: '$31.5',
+      rate: 3,
+      views: '22k',
+      id: 6,
+    },
+  ];
+
+  let createCards = () => {
+    let specialBooks = [];
+    specialBooks = booksArray.slice(0, 4);
+
+    cards = specialBooks.map((book, index) => {
+      return (
+        <MDBCarouselItem
+          itemId={index}
+          key={book.id}
+          className="mt-md-4 mb-md-4 pl-lg-3"
+        >
+          <MDBCard className="w-30 shadow-none" wide>
+            <MDBContainer className="w-full">
+              <MDBRow className="p-0">
+                <MDBCol size="12" md="4" lg="5" className="p-0">
+                  <img
+                    className="imageCardCarsouel"
+                    src={`${book.cover}`}
+                    alt="cover book"
+                  />
+                </MDBCol>
+
+                <MDBCol md="1" lg="1"></MDBCol>
+
+                <MDBCol md="7" lg="6" size="12" className="my-auto">
+                  <MDBCardBody className="p-lg-0 text-center text-md-left text-lg-left mx-auto">
+                    <div>
+                      <h1 className="fontHeader1 mb-0 font-weight-normal  text-2xl">
+                        {' '}
+                        {`${book.authorName}`}{' '}
+                      </h1>
+                    </div>
+
+                    <MDBCardText>
+                      <p className="mb-1 fontHeader2 font-normal font-weight-normal ">{`${book.bookName}`}</p>
+                      <p className="mb-1 parg1 font-weight-normal">{`${book.type}`}</p>
+                      <p className="font-black fontPrice font-weight-normal ">{`${book.price}`}</p>
+
+                      {/* create stars */}
+                      <div className="mb-4">
+                        <div className="inline-block mr-2">
+                          {createStars(book.rate)} ({book.views}) views
+                        </div>
+                      </div>
+                    </MDBCardText>
+
+                    <button className="buttomText text-center mycolorPink w-5/6 shadow-md  hover:shadow-lg h-10 text-black font-bold leading-3 rounded-full">
+                      {t('addtocart')}
+                    </button>
+                  </MDBCardBody>
+                </MDBCol>
+              </MDBRow>
+            </MDBContainer>
+          </MDBCard>
+        </MDBCarouselItem>
+      );
+    });
+
+    return cards;
+  };
+  let createStars = (rate) => {
+    let stars = [];
+    for (let i = 1; i <= 5; i++) {
+      i <= rate
+        ? stars.push(<icon className="fa fa-star text-yellow-300 mr-1 "></icon>)
+        : stars.push(
+            <icon className="fa fa-star text-yellow-300 mr-1 font-light"></icon>
+          );
+    }
```

Comment:
this logic is working but only for the full stars, so imagine the rating was 3.5? what will happen here?
you will need to cover this case also when we have a portion star and try to make a one quarter, half, and three quarters stars and implement them in this code